cc_pam.c

```diff
@@ -45,6 +46,30 @@ PAM_EXTERN int pam_sm_acct_mgmt(pam_handle_t *pamh,
 				int flags, int argc, const char **argv);
 #endif
 
+
+/*
+ * Given the PAM arguments and the user we're authenticating, see if we should
+ * ignore that user because they're root or have a low-numbered UID and we
+ * were configured to ignore such users.  Returns true if we should ignore
+ * them, false otherwise.
+ */
+static int
+_pamcc_should_ignore(const char *username, int minimum_uid)
+{
+	struct passwd *pwd;
+
+	if (minimum_uid > 0) {
+		pwd = getpwnam(username);
```

Comment:
Might be safer to use getpwnam_r() so we don't disturb any state in the calling application.